app/models/invite_redeemer.rb

```diff
@@ -109,7 +115,9 @@ def send_welcome_message
   end
 
   def approve_account_if_needed
-    invited_user.approve(invite.invited_by_id, false)
+    if get_existing_user
```

Comment:
this is the test I was referring to:

rspec ./spec/models/email_token_spec.rb:147 # EmailToken#confirm confirms the token and redeems invite marks the user as approved